### PR TITLE
fix(ios): cancel pending local notifications

### PR DESCRIPTION
**Pull request by janvennemann**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26594

**Optional Description:**
Fix crashes when trying to remove pending local notifications.

### CODE REVIEW

File: iphone/Classes/TiAppiOSProxy.m [hunk 1]
```diff
@@ -955,6 +941,27 @@ - (TiAppiOSLocalNotificationProxy *)scheduleLocalNotification:(id)args
...
 }
 
+/**
+ Assign custom user info to notification content and make sure it has an identifier.
+
+ @param userInfo User info dictionary to assign to the notification content
+ @param content Notification content, can either be UNNotificationContent or UILocalNotification
+ @param notificationIdentifier The unique idenitifer for a notification.
+ */
+- (void)assignUserInfo:(NSDictionary *)userInfo toContent:(id)content ensureIdentifier:(NSString)notificationIdentifier
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiAppiOSProxy.m]: NSString should be  NSString * . It is causing build issue. There are other build issues as well please check.

File: iphone/Classes/TiAppiOSUserNotificationCenterProxy.m [hunk 2]
```diff
@@ -101,18 +101,17 @@ - (void)removePendingNotifications:(id)args
       }
 
       [center getPendingNotificationRequestsWithCompletionHandler:^(NSArray<UNNotificationRequest *> *requests) {
-        // Loop through current notification requests
+        NSMutableArray<NSString *> *identifiers = [NSArray new];
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiAppiOSUserNotificationCenterProxy.m]: [NSArray new] should be [NSMutableArray new].

File: iphone/Classes/TiAppiOSUserNotificationCenterProxy.m [hunk 3]
```diff
@@ -101,18 +101,17 @@ - (void)removePendingNotifications:(id)args
...
         for (UNNotificationRequest *request in requests) {
-
-          // Loop through provided notifications
           for (id notification in args) {
-            ENSURE_TYPE(notification, TiAppiOSLocalNotificationProxy);
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiAppiOSUserNotificationCenterProxy.m]: Rather removing this we should use ENSURE_TYPE(notification, NSDictionary).

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]

**Comment by vijaysingh-axway** [comment]
@janvennemann  While scanning the class TiAPPiOSUserNotificationCenterProxy, I found that method
- (void)removeDeliveredNotifications:(id)args has similar problem as of method
- (void)removePendingNotifications:(id)args.

Please fix that as well. Thanks!

File: iphone/Classes/TiAppiOSUserNotificationCenterProxy.m [hunk 4]
```diff
@@ -145,19 +152,20 @@ - (void)removeDeliveredNotifications:(id)args
         return;
       }
 
-      [center getPendingNotificationRequestsWithCompletionHandler:^(NSArray<UNNotificationRequest *> *requests) {
-        // Loop through current notification requests
+      [center getDeliveredNotificationRequestsWithCompletionHandler:^(NSArray<UNNotificationRequest *> *requests) {
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiAppiOSUserNotificationCenterProxy.m]: @janvennemann  In callback this method returns " NSArray<UNNotification *> * _Nonnull notifications" rather  "NSArray<UNNotificationRequest *> *requests" . See doc [here](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/1649520-getdeliverednotificationswithcom?language=objc). Please update the implementation accordingly. Thanks!

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiAppiOSUserNotificationCenterProxy.m (hunk 4 above)
Comment (janvennemann) [reply iphone/Classes/TiAppiOSUserNotificationCenterProxy.m]: Oh, the method name was also wrong, fixed the issues!

File: iphone/Classes/TiAppiOSUserNotificationCenterProxy.m [hunk 5]
```diff
@@ -47,7 +47,7 @@ - (void)getPendingNotifications:(id)callback
     NSMutableArray *result = [NSMutableArray arrayWithCapacity:[notifications count]];
 
     for (UILocalNotification *notification in notifications) {
-      [result addObject:[TiApp dictionaryWithLocalNotification:notification withIdentifier:nil]];
+      [result addObject:[TiApp dictionaryWithLocalNotification:notification]];
```
Comment (vijaysingh-axway) [inline iphone/Classes/TiAppiOSUserNotificationCenterProxy.m]: @janvennemann Everything looks good. I can see warning for [TiApp dictionaryWithLocalNotification:notification]. Reason is  `+ (NSDictionary *)dictionaryWithLocalNotification:(UILocalNotification *)notification` method is not exposed in header file. So we should use `[TiApp dictionaryWithLocalNotification:notification withIdentifier:notification.userInfo[@"id"]]`. 
After fixing this we are good to go.
Thanks!

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]

File: iphone/Classes/TiAppiOSUserNotificationCenterProxy.m (hunk 5 above)
Comment (janvennemann) [reply iphone/Classes/TiAppiOSUserNotificationCenterProxy.m]: Fixed!

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Review by ssjsamir — APPROVED** [review]
**FR Passed** Using the following test case
```
var window = Ti.UI.createWindow({
  backgroundColor: 'white'
});

var button = Ti.UI.createButton({
  title: 'Schedule'
});

var isScheduling = true;
button.addEventListener('click', function() {
  if (isScheduling) {
    var alarmObject = {
      date: new Date(new Date().getTime() + 20000),
      alertBody: "Hello world",
      badge: 1,
      identifier: "test_id",
      sound: 'default'
    };

    console.log('schedule: ' + JSON.stringify(alarmObject));

    Ti.App.iOS.scheduleLocalNotification(alarmObject);

    button.title = "Cancel";
  } else {
    Ti.App.iOS.UserNotificationCenter.getPendingNotifications(function(result) {
      if (result && result.notifications && result.notifications.length) {
        console.log(result.notifications);
        var _;
        var notificationsToCancel = result.notifications

        console.log('cancel these: ' + JSON.stringify(notificationsToCancel));

        if (notificationsToCancel) {
          Ti.App.iOS.UserNotificationCenter.removePendingNotifications(notificationsToCancel);
        }
      }
    });

    button.title = "Schedule";
  }

  isScheduling = !isScheduling;
});

window.add(button);

window.addEventListener('open', function() {
  Ti.App.iOS.registerUserNotificationSettings({
    types: [
      Ti.App.iOS.USER_NOTIFICATION_TYPE_ALERT,
      Ti.App.iOS.USER_NOTIFICATION_TYPE_SOUND,
      Ti.App.iOS.USER_NOTIFICATION_TYPE_BADGE
    ]
  });
});

window.open();

```
Application no longer shows an error unlike 7.5.0.GA.

**Test Environment**
```
iPhone 6s plus (9.3) Sim
APPC CLI: 7.0.0
Operating System Name: Mac OS Mojave
Operating System Version: 10.14.2
Node.js Version: 8.9.1
Xcode 10.1
```

**Pull request merged**